From fe33f1ca460772f19c0564ae1a36d3375b739b21 Mon Sep 17 00:00:00 2001
From: Makoto Onuki <omakoto@google.com>
Date: Wed, 10 Feb 2021 12:16:06 -0800
Subject: [PATCH] Add SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS to
 CalendarDebugActivity

Bug: 174046397
Test: manual test
Change-Id: Id34eaaf1a8beab80d768186a271201e0b58d4a19
(cherry picked from commit 11fe2048a5aa5d3d3db315194921130bf2407919)
---
 AndroidManifest.xml                                        | 1 +
 .../android/providers/calendar/CalendarDebugActivity.java  | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 9d87918..fd56187 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -18,6 +18,7 @@
         package="com.android.providers.calendar"
         android:sharedUserId="android.uid.calendar">

+    <uses-permission android:name="android.permission.HIDE_NON_SYSTEM_OVERLAY_WINDOWS" />
     <uses-permission android:name="android.permission.READ_CALENDAR" />
     <uses-permission android:name="android.permission.WRITE_CALENDAR" />
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
diff --git a/src/com/android/providers/calendar/CalendarDebugActivity.java b/src/com/android/providers/calendar/CalendarDebugActivity.java
index d06e095..70a1fe4 100644
--- a/src/com/android/providers/calendar/CalendarDebugActivity.java
+++ b/src/com/android/providers/calendar/CalendarDebugActivity.java
@@ -69,6 +69,13 @@ public class CalendarDebugActivity extends Activity implements OnClickListener {
         updateDeleteButton();
     }

+    @Override
+    protected void onStart() {
+        super.onStart();
+        getWindow().addSystemFlags(android.view.WindowManager.LayoutParams
+                .SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+    }
+
     private void updateDeleteButton() {
         final boolean fileExist =
             new File(Environment.getExternalStorageDirectory(), OUT_FILE).exists();
--
2.29.0
